### Personal Statement 
A self-taught Computational Biomedical professional with background in Molecular Medicine and an extensive experience in Biomedical Research. 

### Education
Molecular Medicine and Translational Research, M.Sc. 

### Computational Skills
R, Linux, Python
 
### Projects:

## RNAseq reads processing pipeline for human Breast Cancer cells

> - [Terminal commands](https://github.com/amenamahdami/Amena_Mahdami_Portfolio.github.io/blob/main/Shell/terminal%20commands.sh)
> - [Bash Scripts: RNAseq Pipeline](https://github.com/amenamahdami/Amena_Mahdami_Portfolio.github.io/blob/main/Shell/RNASeq_pipeline_BC.sh)

- RNAseq data generated by S Ishihara, et al. Sci Rep 2022 ([PMID: 35428832](https://pubmed.ncbi.nlm.nih.gov/35428832/) , [GEO: GSE186211](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE186211))
- In this study they show that the lactate G protein-coupled receptor 81 (GPR81) regulates the malignant phenotype of breast cancer cell by reprogramming energy metabolism.
- To further determine the role of GPR81 in breast cancer, GPR81 was stably silenced using GPR81-specific short hairpin (sh)RNA in human breast cancer MDA-MB-231 cells, which showed high expression of GPR81. Silenced cells were compared with cells expressing the control hairpin RNA (shNT). And thereafter, RNA-seq was performed in extracted RNA of those samples.  

- In this project I first extract the fastq files from the SRA Database using SRA toolkit.
- In the second part, I process the reads in the fastq files to get the RNA-seq counts matrix.
- RNAseq pipeline steps:
  1. Run QC on read using FastQC 
  2. Mapping of reads to genome using HISAT2
  3. Reads quantification using featureCounts![image](https://github.com/amenamahdami/Amena_Mahdami_Portfolio.github.io/assets/117296183/65d58fe7-6dbf-4713-9ef3-580d2aa59198)
![image](https://github.com/amenamahdami/Amena_Mahdami_Portfolio.github.io/assets/117296183/5315b86f-3cfb-4b64-8b5a-6783b40c7609)
![image](https://github.com/amenamahdami/Amena_Mahdami_Portfolio.github.io/assets/117296183/a1560c75-e64f-4d8c-9ac8-eabc0e589e65)

    
- Results from the run: [RNA-seq counts matrix](https://github.com/amenamahdami/Amena_Mahdami_Portfolio.github.io/blob/main/Shell/sample_counts.tsv) , [test_counts.summary](https://github.com/amenamahdami/Amena_Mahdami_Portfolio.github.io/blob/main/Shell/test_counts.summary)  

    

## Differential Gene Expression Analysis using DESeq2 for human primary Airway Smooth Muscle cells
> - R Scripts: [Getting Data](https://github.com/amenamahdami/Amena_Mahdami_Portfolio.github.io/blob/main/R/getData_airway.r) , [Running Analysis](https://github.com/amenamahdami/Amena_Mahdami_Portfolio.github.io/blob/main/R/DESeq2_workflow_airway.R)  

- RNAseq data generated by BE Himes, et al. PLoS One 2014 ([PMID: 24926665](https://pubmed.ncbi.nlm.nih.gov/24926665/) , [GEO: GSE52778](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52778))
- In this project I use RNA-seq counts matrix from smooth muscle cells to identify genes that have differential gene expression patterns.
- RNAseq performed in human primary Airway Smooth Muscle cell lines which are treated with Dexamethasone (1 ÂµM for 18 hours), which is used for asthma patients to reduce inflammation in the airways.
- Objective of project: Characterize Differential Gene Expressions between treated and untreated, to understand the transcriptional changes occurring due to treatment with Dexamethasone.
- Workflow steps:
  1. Read the data and get it in the right format
  2. Create DESeq dataset object
  3. Run the DESeq()
  4. Explore and visualize results

![Differential Gene Expression Volcano Plot](assets/img/Volcano_plot_airway.png)  

  
  
## Heart Disease Prediction
> - [Python Script](https://github.com/amenamahdami/Amena_Mahdami_Portfolio.github.io/blob/main/Python/ML_Project_Heart_Disease_Prediction.ipynb)

- In this project I use publicly available Heart Disease Dataset downloaded from [Kaggle](https://www.kaggle.com/datasets/johnsmith88/heart-disease-dataset), to predict heart disease incident.
- The first part of the project I perform Data Wrangling and Exploration.
- In the second part I use Machine Learning to predict whether a patient is having high risk of heart disease or not.


      


### Resources, References & Training:
- [HISAT2 Manual](https://daehwankimlab.github.io/hisat2/manual/)
- [featureCounts](https://rnnh.github.io/bioinfo-notebook/docs/featureCounts.html)
- [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
- [Linux/Unix Command Cheat Sheet](https://rumorscity.com/2014/08/16/6-best-linuxunix-command-cheat-sheet/)
- [Bioconductor](https://www.bioconductor.org/)
- [DESeq2 package](https://bioc.ism.ac.jp/packages/2.14/bioc/vignettes/DESeq2/inst/doc/beginner.pdf)
- [DESEQ2 R Tutorial](https://lashlock.github.io/compbio/R_presentation.html)
- [Python Data Science Handbook](https://jakevdp.github.io/PythonDataScienceHandbook/)
- [scikit-learn Machine Learning in Python](https://scikit-learn.org/stable/)
- [scikit-learn Classifier comparison](https://scikit-learn.org/stable/auto_examples/classification/plot_classifier_comparison.html#sphx-glr-auto-examples-classification-plot-classifier-comparison-py)
- [Introduction to Bioinformatics and Computational Biology course](https://liulab-dfci.github.io/bioinfo-combio/)
- [Google Data Analytics Certificate](https://coursera.org/share/0dd196ce17876b5d71ccc0c4695b738f) 
